course_catalog/etl/loaders.py

```diff
@@ -0,0 +1,55 @@
+"""Course catalog data loaders"""
+from django.contrib.contenttypes.models import ContentType
+
+from course_catalog.models import Course, Program, ProgramItem
+
+
+def load_course(course_data):
+    """Load the course into the database"""
+    course_id = course_data.pop("course_id")
+    course, _ = Course.objects.update_or_create(
+        course_id=course_id, defaults=course_data
+    )
+
+    return course
+
+
+def load_program(program_data):
+    """Load the program into the database"""
+    program_id = program_data.pop("program_id")
+    courses_data = program_data.pop("courses")
+
+    program, _ = Program.objects.update_or_create(
+        program_id=program_id, defaults=program_data
+    )
+
+    courses = []
+    course_content_type = ContentType.objects.get(model="course")
+
+    for position, course_data in enumerate(courses_data):
+        # skip courses that don't define a course_id
+        if not course_data.get("course_id", None):
+            continue
+
+        course = load_course(course_data)
```

Comment:
if you add this code just before calling `load_course`:
```python
        course_data.update({
            "program_type": "MicroMasters",
            "program_name": program.title,
            "offered_by": OfferedBy.micromasters.value
        })
```
then you could get rid of the following:
- the [micromasters code block](https://github.com/mitodl/open-discussions/blob/28c4e91263e65558848325587e96f581d0feb356/course_catalog/api.py#L440-L447) in `course_catalog.api.tag_edx_program`
- the `course_catalog.api.get_micromasters_data` function
- `MICROMASTERS_COURSE_URL` setting and env variable

Reply:
I think it might be best to leave this as-is for now so this PR stays additive in nature and because we need to be able to support #2163 in order to do that and we aren't tackling that yet. I'm also not sure having those `program_*` fields on the course is right given how our data is structured, but I think that requires further digging. Does that seem right?

Reply:
ok, it just seems a little weird to update courses with data from 2 different MM URL's using separate functions sharing the same name, but it still works, I'm fine with optimizing this in a future PR.

Reply:
Yeah, I agree, it's just sort of a slippery slope I think in terms of what'd get rolled into this change...